File: 0048-rotate-image/0048-rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        for(int i = 0; i < matrix.size(); i++){
            for(int j = 0; j < matrix[0].size(); j++){
                if(matrix[i][j] > 0){
                    int save = matrix[i][j]; 
                    matrix[i][j] = matrix[j][i] * -1; 
                    matrix[j][i] = save * -1; 
                   // matrix[i][j] *= -1; 
                    //matrix[j][i] *= -1; 
                    //swap(matrix[i][j], matrix[j][i]); 
                }
            }
        }

        for(int i = 0; i < matrix.size(); i++){
            for(int j = 0; j < matrix[i].size(); j++){
                //if(matrix[i][j] < 0){
                matrix[i][j] *= -1; 
                //}
            }
        }

        for(int i = 0; i < matrix.size(); i++){
            for(int j = 0; j < matrix.size() / 2; j++){
                swap(matrix[i][j], matrix[i][matrix.size() - j - 1]); 
            }
        }
    }
};

File: 0048-rotate-image/0048-rotate-image.java
class Solution {
    public void rotate(int[][] matrix) {
        for(int i = 0; i < matrix.length; i++){
            for(int j = i + 1; j < matrix.length; j++){
                int tmp = matrix[i][j]; 
                matrix[i][j] = matrix[j][i]; 
                matrix[j][i] = tmp; 
            }
        }
        for(int i = 0; i < matrix.length; i++){
            reverse(matrix[i]); 
        }
    }
    public void reverse(int[] matrix){
        int start = 0, end = matrix.length-1; 
        while(start < end){
            int tmp = matrix[start]; 
            matrix[start] = matrix[end]; 
            matrix[end] = tmp; 
            start++;
            end--; 
        }
    }
}